### PR TITLE
hist plot: do not show negative y range

### PR DESCRIPTION
**Pull request by jgehrcke**
For issue #1370.

Before:

Screenshot from 2023-07-21 13-58-58

After:

Screenshot from 2023-07-21 14-05-59

### CODE REVIEW

File: conbench/app/_plots.py [hunk 1]
```diff
@@ -887,6 +887,12 @@ def time_series_plot(
     p.legend.location = "top_left"
     p.legend.label_text_font_size = "12px"
 
+    # y range should not go into the negative, but it should also not
+    # always start at 0. Inspect `source_rolling_alert_min_over_time` which
+    # should conceptually hold the minimal value across all sources.
+    if min(source_rolling_alert_min_over_time.data["y"]) < 0:
+        p.y_range.start = 0
```
Comment (jgehrcke) [inline conbench/app/_plots.py]: creative solution with the help of
https://github.com/bokeh/bokeh/issues/8421
https://stackoverflow.com/a/61119600
https://discourse.bokeh.org/t/axis-range-cannot-be-changed-in-bokeh-dashboard-if-data-updated/10284/2

**Comment by jgehrcke** [comment]
:ship:, let's get this in

**Comment by alistaire47** [comment]
This is good, but we should still `np.clip()` the bounds to 0 since negative bounds are statistically invalid. And/or calculating bounds asymmetrically, since distributions will frequently be skew.

**Comment by deepsihag** [comment]
In case of `more-is-better` would this still be true about not showing negative ?

**Comment by alistaire47** [comment]
Those are rates (bytes/second or iterations/second), which can't be negative either. Though it arguably may not be sensible for the y-axis to go all the way to 0 for them.

**Comment by deepsihag** [comment]
Ah ok.. make sense.  I missed looking at values and was just assuming the negative/lower side of mean but this is negative value.

**Pull request merged**